### PR TITLE
Project Round Record

### PR DESCRIPTION
**Pull request by aminlatifi**

### CODE REVIEW

File: src/repositories/projectRoundRecordRepository.test.ts [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        earlyAccessRound1.id,
+      );
+
+      expect(records).to.have.lengthOf(1);
+      expect(records[0].totalDonationAmount).to.equal(donationAmount);
+      expect(records[0].totalDonationUsdAmount).to.equal(donationUsdAmount);
+    });
+  });
+
+  describe('getProjectRoundRecord test cases', () => {
```
Comment (ae2079) [inline src/repositories/projectRoundRecordRepository.test.ts:260]:
```suggestion
  describe('getCumulativePastRoundsDonationAmounts test cases', () => {
```

File: src/entities/projectRoundRecord.ts [hunk 2]
```diff
@@ -6,32 +6,38 @@ import {
...
   totalDonationAmount: number;
 
   @Field(_type => Float)
-  @Column({ type: 'real', default: 0 })
+  @Column({ type: 'float', default: 0 })
   totalDonationUsdAmount: number;
 
+  @Field(_type => Float, { nullable: true })
+  @Column({ type: 'float', nullable: true })
+  cumulativePastRoundsDonationAmounts?: number | null;
```
Comment (ae2079) [inline src/entities/projectRoundRecord.ts:32]: We don't fill this field anywhere, just read it in the `getCumulativePastRoundsDonationAmounts` function!

**Review by ae2079 — CHANGES_REQUESTED** [review]
Thanks @aminlatifi

File: migration/1727292637656-addEarlyAccessIndex.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddEarlyAccessIndex1727292637656 implements MigrationInterface {
+  name = 'AddEarlyAccessIndex1727292637656';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "early_access_round" DROP CONSTRAINT "UQ_e2f9598b0bbed3f05ca5c49fedc"`,
```
[the remark this answers is not in the document]
Comment (aminlatifi) [reply migration/1727292637656-addEarlyAccessIndex.ts:8]: It's generated automatically. I made an index explicitly and made it uniq

File: src/entities/projectRoundRecord.ts (hunk 2 above)
Comment (aminlatifi) [reply src/entities/projectRoundRecord.ts:32]: We must fill it in an appropriate place, must figure out when and where is the beset

**Pull request merged**